### PR TITLE
Add pyflakes and make corrections accordingly

### PR DESCRIPTION
**Pull request by loicgasser**
Pyflakes checks for various errors in the source code.

This PR fixes errors such as
* module imported but unused
* from module import *’ used; unable to detect undefined names
* redefinition of unused name from line N
* list comprehension redefines name from line N
* undefined name name
* local variable name is assigned to but never used

While pep8 checks styling, pyflakes checks code integrity.

This even fixes some bugs.

### CODE REVIEW

**Comment by loicgasser** [comment]
test again

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

File: buildout.cfg [hunk 1]
```diff
@@ -223,9 +224,12 @@ on_update = true
...
+    FILES=$(find ${validate-py:folders} -name \*.py)
+    FIXME+=$(echo $FILES | xargs ./buildout/bin/pep8 --ignore=E501)
+    FIXME+=$(echo -e "\n ")
+    FIXME+=$(echo $FILES | xargs ./buildout/bin/pyflakes)
 
-    if [[ $FIXME == chsdi* || $FIXME == mapproxy* ]];
+    if [[ $FIXME == *"chsdi"* || $FIXME == *"mapproxy"* ]];
     then
       echo "$(tput setaf 1)$FIXME$(tput sgr0)" &&
       echo "You can fix automatically some styling errors using the following command:" &&
```
Comment (gjn) [inline buildout.cfg]: Does pyflakes have an autocorrect mode as pep? The message here is pep specific only.

**Comment by gjn** [comment]
I think this is good to go. I'll merge.

**Pull request merged**